### PR TITLE
otelslog: Add example

### PR DESCRIPTION
**Pull request by pellared**
Towards https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5186

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: bridges/otelslog/exampe_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"go.opentelemetry.io/contrib/bridges/otelslog"
+	"go.opentelemetry.io/otel/log/noop"
+)
+
+func Example() {
+	// Use a working LoggerProvider implementation instead e.g. using go.opentelemetry.io/otel/sdk/log.
+	provider := noop.NewLoggerProvider()
+
+	// Create an *slog.Logger with *otelslog.Handler and use it in your application.
+	slog.New(otelslog.NewHandler(otelslog.WithLoggerProvider(provider)))
```
Comment (dashpole) [inline bridges/otelslog/exampe_test.go:18]: Should we use otelslog.NewLogger instead to simplify it?
```suggestion
	otelslog.NewLogger(otelslog.WithLoggerProvider(provider))
```

Comment (pellared) [reply bridges/otelslog/exampe_test.go:18]: I propose `otelslog.NewHandler` as most (all?) packages I am aware have only `Handler` factories. The reason is that one may want to use it together with some handler decorator e.g. https://github.com/samber/slog-multi. The `slog` package has `NewJSONHandler` and `NewTextHandler`.  See also: https://go.dev/wiki/Resources-for-slog

I even propose removing `otelslog.NewLogger` for sake of consistency and in order to remove redundant functionality.

@MrAlias WDYT?

Comment (dashpole) [reply bridges/otelslog/exampe_test.go:18]: See https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5292 for the previous discussion.  I agree with the need for NewHandler for decorators.  I like the simplicity of being able to do `slog.SetDefault(otelslog.NewLogger())`, but I don't think NewLogger is strictly necessary.

Comment (MrAlias) [reply bridges/otelslog/exampe_test.go:18]: I think @dashpole is correct, the simplicity of `slog.SetDefault(otelslog.NewLogger())` is likely what most users are going to want.

Comment (pellared) [reply bridges/otelslog/exampe_test.go:18]: Personally, I lean towards smaller API surface. But it is just a preference.

Comment (pellared) [reply bridges/otelslog/exampe_test.go:18]: Applying suggestion and resolving

**Comment by dashpole** [comment]
Should the example show how to call SetDefault?

**Comment by pellared** [comment]
> Should the example show how to call slog.SetDefault?

I did not want to do that as many people find global logger as an anti-pattern.

**Review by XSAM — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**